libraries/GCS_MAVLink/GCS_Common.cpp

```diff
@@ -211,6 +211,12 @@ void GCS_MAVLINK::send_mcu_status(void)
 }
 #endif
 
+// returns the battery remaining percentage if valid, -1 otherwise
+int8_t GCS_MAVLINK::battery_remaining_pct(const uint8_t instance) const {
+    uint8_t percentage;
+    return AP::battery().capacity_remaining_pct(percentage) ? MIN(percentage, INT8_MAX) : -1;
```

Comment:
Bug here. We need to get each instance's % remaining

```suggestion
    return AP::battery().capacity_remaining_pct(percentage, instance) ? MIN(percentage, INT8_MAX) : -1;
```

Reply:
Shouldn't we do that in another PR? Something like "getting each instance battery capacity".

Reply:
No. See L273 of the old code.  

https://github.com/ArduPilot/ardupilot/blob/f606fe727a83f336573a292e1405acf21ef9e964/libraries/GCS_MAVLink/GCS_Common.cpp#L272-L273

We were sending out the battery percentage for each instance in `send_battery_status`. With what you have here we only ever send out the primary instance's percentage incorrectly to every instance. 

In the other messages they do not have a notion of different battery instances and thus ask to get the primary instance.

Reply:
Oops. Sorry, I missed it. Fixed it now. Thanks for correcting me.